mobile/examples/speech_recognition/ios/SpeechRecognition/SpeechRecognizer.swift

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import Foundation
+
+private let kTokens = [
+  "<s>", "<pad>", "</s>", "<unk>", "|", "E", "T", "A", "O", "N", "I", "H", "S", "R", "D", "L", "U", "M", "W", "C", "F",
+  "G", "Y", "P", "B", "V", "K", "'", "X", "J", "Q", "Z",
+]
+
+class SpeechRecognizer {
+  private let ortEnv: ORTEnv
+  private let ortSession: ORTSession
+
+  enum SpeechRecognizerError: Error {
+    case Error(_ message: String)
+  }
+
+  init() throws {
+    ortEnv = try ORTEnv(loggingLevel: ORTLoggingLevel.warning)
+    guard let modelPath = Bundle.main.path(forResource: "wav2vec2-base-960h.all", ofType: "ort") else {
+      throw SpeechRecognizerError.Error("Failed to find model file.")
+    }
+    ortSession = try ORTSession(env: ortEnv, modelPath: modelPath, sessionOptions: nil)
+  }
+
+  private func postprocess(modelOutput: UnsafeBufferPointer<Float>) -> String {
+    func maxIndex(_ values: UnsafeBufferPointer<Float>) -> Int? {
+      var max: (idx: Int, value: Float)?
+      for (idx, value) in values.enumerated() {
+        if max == nil || value > max!.value {
+          max = (idx, value)
+        }
+      }
+      return max?.idx
+    }
+
+    func tokenIndexToOutput(_ index: Int) -> String {
+      if index == 4 {
+        return " "
+      } else if index > 4 && index < kTokens.count {
+        return kTokens[index]
+      }
+      return ""
+    }
+
+    let n = modelOutput.count / kTokens.count
+    var resultTokenIndices: [Int] = []
+
+    for i in 0..<n {
+      let tokenValues = UnsafeBufferPointer<Float>(rebasing: modelOutput[i * kTokens.count..<(i + 1) * kTokens.count])
+      if let tokenIndex = maxIndex(tokenValues) {
+        // append without consecutive duplicates
+        if tokenIndex != resultTokenIndices.last {
+          resultTokenIndices.append(tokenIndex)
+        }
+      }
+    }
+
+    return resultTokenIndices.map(tokenIndexToOutput).joined()
+  }
+
+  func evaluate(inputData: Data) throws -> String {
+    let inputShape: [NSNumber] = [1, inputData.count / MemoryLayout<Float>.stride as NSNumber]
+    let input = try ORTValue(
+      tensorData: NSMutableData(data: inputData),
+      elementType: ORTTensorElementDataType.float,
+      shape: inputShape)
+
+    let startTime = DispatchTime.now()
+    let outputs = try ortSession.run(
+      withInputs: ["input": input],
+      outputNames: ["output"],
+      runOptions: nil)
+    let endTime = DispatchTime.now()
+    print("ORT session run time: \(Float(endTime.uptimeNanoseconds - startTime.uptimeNanoseconds) / 1.0e6) ms")
```

Comment:
Is the inference time info intended to be displayed? If so, we can use `Inference Time: X ms` to be consistent with the other sample applications.
(didn't see `inference time` is included in the screenshot though).

Reply:
it's not intended to be displayed now, just a debug print